File: python/IsomorphicStrings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        d, seen = {}, set()
        for i in range(len(s)):
            if s[i] not in d and t[i] not in seen:
                d[s[i]] = t[i]
                seen.add(t[i])
            if d.get(s[i]) != t[i]:
                return False
        return True

File: C/IsomorphicStrings.c
#include <stdbool.h>

bool isIsomorphic(char* s, char* t)
{
	int hash[128] = {0}, seen[128] = {0};  //already matched
	for (int i = 0; s[i]; ++i)
	{
		if (hash[s[i]] == 0 && seen[t[i]] == 0)
		{
			hash[s[i]] = t[i];
			seen[t[i]] = 1;
		}
		if (hash[s[i]] != t[i])
			return false;
	}
	return true;
}
